File: Leetcode/1365_smaller_than_current_number/solution.go
package main

import "fmt"

func smallerThanCurrentNumber(nums []int) []int {

	result := make([]int, len(nums))

	for i := 0; i < len(nums); i++ {
		count := 0;
		for j := 0; j < len(nums); j++ {
			if nums[j] < nums[i] {
				count++;
			}
		}
		result[i] = count;
	}

	return result;

}

func main() {
	nums := []int {7,7,7,7};

	fmt.Println(smallerThanCurrentNumber((nums)))
}

File: Leetcode/1365_smaller_than_current_number/solution.py


def smallerThanCurrentNumber(nums):

    newNums = sorted(nums)
    numLen = len(nums)

    output = [0] * numLen

    d = {}

    for n in range(numLen):
        if newNums[n] in d:
            continue
        else:
            d[newNums[n]] = n

    for i in range(numLen):
        output[i] = d[nums[i]]

    return output


# nums = [8, 1, 2, 2, 3]
nums = [6, 5, 4, 8]
# [4, 5, 6, 8]
print(smallerThanCurrentNumber(nums))
